dlpy/tests/test_model.py

```diff
@@ -1039,6 +1043,422 @@ def test_mobilenetv2(self):
                                        offsets=255*[0.485, 0.456, 0.406],
                                        norm_stds=255*[0.229, 0.224, 0.225])
 
+    def test_fit_tensorboard(self):
+        try:
+            import tensorflow as tf
+            import numpy as np
+            from tensorflow.core.util import event_pb2
+        except:
+            unittest.TestCase.skipTest(self, "tensorflow and/or np not found in the libraries")
+
+        if self.data_dir is None:
+            unittest.TestCase.skipTest(self, "DLPY_DATA_DIR is not set in the environment variables")
+
+        if os.path.exists(self.data_dir + '_TB'):
+            log_dir = self.data_dir + '_TB'
+        else:
+            os.mkdir(self.data_dir + '_TB')
+            log_dir = self.data_dir + '_TB'
+
+        caslib, path, tmp_caslib = caslibify(self.s, path=self.data_dir+'images.sashdat', task='load')
+        self.s.table.loadtable(caslib=caslib, casout={'name': 'eee', 'replace': True}, path=path)
+
+        model0 = Sequential(self.s, model_table='Simple_CNN0')
+        model0.add(InputLayer(3, 224, 224))
+        model0.add(Conv2d(8, 7))
+        model0.add(Pooling(2))
+        model0.add(Conv2d(8, 7))
+        model0.add(Pooling(2))
+        model0.add(Dense(16))
+        model0.add(OutputLayer(act='softmax', n=2))
+
+        # Run without TB
+        r = model0.fit_tensorboard(data='eee', inputs='_image_', target='_label_', lr=0.0001)
+        self.assertTrue(r.severity <= 1)
+
+        # Run model1 with TB no valid - 5 epochs
+        model1 = Sequential(self.s, model_table='Simple_CNN1')
+        model1.add(InputLayer(3, 224, 224))
+        model1.add(Conv2d(8, 7))
+        model1.add(Pooling(2))
+        model1.add(Conv2d(8, 7))
+        model1.add(Pooling(2))
+        model1.add(Dense(16))
+        model1.add(OutputLayer(act='softmax', n=2))
+
+        tensorboard = TensorBoard(model1, log_dir)
+        r = model1.fit_tensorboard(data='eee', inputs='_image_', target='_label_', lr=0.0001, tensorboard=tensorboard)
+        
+        epochs = model1.training_history.Epoch.values
+        lr = model1.training_history.LearningRate.values
+        loss = model1.training_history.Loss.values
+        error = model1.training_history.FitError.values
+        self.assertTrue(len(epochs) == 5)
+
+        # Check number of event files created (1 for each call to fit)
+        self.assertTrue(len(os.listdir(log_dir +'Simple_CNN1' + '/learning_rate/')) == 1)
```

Comment:
Just wondering about all of the '/' path separators here.  Are we guaranteed to be running in an environment that is UNIX-only?